src/NuGet.Clients/VsExtension/NuGetPackage.cs

```diff
@@ -390,10 +390,10 @@ private IEnumerable<NuGet.Credentials.ICredentialProvider> GetCredentialProvider
                 VisualStudioAccountProvider.FactoryMethod,
                 errorDelegate);
 
-            var vstsProvider = importer.GetProvider();
-            if (vstsProvider != null)
+            var importedProviders = importer.GetProviders();
+            if (importedProviders.Count != 0)
             {
-                credentialProviders.Add(vstsProvider);
+                credentialProviders.AddRange(importedProviders);
             }
 
             var webProxy = (IVsWebProxy)GetService(typeof(SVsWebProxy));
```

Comment:
Is the IEnumerable<ICredentialProvider> returned by this method supposed to be respecting any order?

Sure, I can move the proxy provider up. The order before this change was:

1. SettingsCredentialProvider
1. vsts provider when available (which in this change is an ImportMany which will include the vsts provider when available)
1. proxy provider

You want the proxy to be first, or right after the settings provider?